packages/core/src/execution-engine/partial-execution-utils/rewire-graph.ts

```diff
@@ -0,0 +1,29 @@
+import * as a from 'assert/strict';
+import { type INode, NodeConnectionTypes } from 'n8n-workflow';
+
+import { type DirectedGraph } from './directed-graph';
+
+export function rewireGraph(tool: INode, graph: DirectedGraph): DirectedGraph {
+	graph = graph.clone();
```

Comment:
Nitpick comment: I usually treat function arguments as immutable, helps to lower complexity. But we don't have the `no-param-reassign` eslint rule enabled, so I'm not sure what our policy on that is :)